### PR TITLE
Change logic for determining minimum polling frequency.

### PR DESCRIPTION
**Pull request by rjnagal**
Since polling is tied to housekeeping, minimum supported polling
frequency is 1s.

Users can specify polling frequency higher than 1s. The polling loop
will be called at the minimum frequency specified in config as long as
its higher than the minimum supported frequency.

### CODE REVIEW

**Comment by rjnagal** [comment]
@anushree-n  I think this is the model we want to go with. WDYT?

File: collector/generic_collector.go [hunk 1]
```diff
@@ -85,6 +82,12 @@ func NewCollector(collectorName string, configfile string) (*GenericCollector, e
 		}
 	}
 
+	// Minimum supported polling frequency is 1s.
+	minSupportedFrequency := 1 * time.Second
```
Comment (vmarmol) [inline collector/generic_collector.go]: In theory it should be the value of minHousekeeping flag :)

Comment (rjnagal) [inline collector/generic_collector.go]: yes, but importing HousekeepingInterval adds a cyclic dependency between manager and collector. I'll clean that up next.

**Comment by anushree-n** [comment]
I believe the earlier model took the minimum of next stats collection and next custom metrics collection as the next housekeeping time.

If the minimum supported frequency is 1s, yes, this looks good! :)

**Comment by rjnagal** [comment]
yeah, the intention is to not go below the housekeeping interval.

**Comment by rjnagal** [comment]
self-merging ....

**Pull request merged**